### ISSUE TITLE
Android notification always appear to show the default ic_popup_reminder (bell-like) icon regardless

### ISSUE THREAD

**Issue by gkimpson** [katzer/cordova-plugin-local-notifications#890]
I have setup the local notifications and they work really well with the Ionic framework. Within my Ionic Ready block I have the following code in which I have set the icon image (I am using this image http://sciactive.com/pnotify/includes/github-icon.png)

As the path is relative to the app.js file, the image is up a level within a folder called 'img' and the file is called 'github-icon.png'.

For some reason the same 'bell' default icon always appears?? Is this a bug? I am using Android 6.0.1 on my Nexus 5.

```
$scope.scheduleSingleNotification = function () {
      $cordovaLocalNotification.schedule({
        id: 1,
        title: 'Warning',
        text: 'my first notification!',
        icon: 'file://img/logo.png',
      }).then(function (result) {
        console.log('Notification #1 triggered');
      });
    };
```

https://github.com/katzer/cordova-plugin-local-notifications/wiki/10.-URIs
This is the image that appears in my local notification
ftp://ftp.gunadarma.ac.id/android/sdk/sdk_310712/docs/resources/samples/ApiDemos/res/drawable-hdpi/ic_popup_reminder.html

**Comment by mix3d** [comment]
Because the Android native code part of this plugin is looking for the asset, if you read the docs it says you have to use the `res://filenamewithoutextension`, and that it has to be in the `res/drawable` folder, not in the www folder.

**Comment by bluesliverx** [comment]
After digging into the code, I finally found out why exactly it doesn't work the way you'd expect.  The docs should be improved (if I get a second I'll submit a PR for the wiki):

* Small icons support resource files ONLY (see the javadocs here: http://developer.android.com/reference/android/support/v4/app/NotificationCompat.Builder.html#setSmallIcon(int) - note that the plugin uses NotificationCompat.Builder and not Notification.Builder for backwards support)
* Large icons support bitmaps ONLY, but you can load the bitmap from a resource file or a URI (see javadocs again)

Because of these, the code is structured to check if the small icon is present.  If it is not, it will attempt to set the small icon from the "icon" config param (only ends up using a resource file).  If it is present, it will attempt to set the small and large icons as configured with large using the "icon" config param and supporting URI.

I'm going to test out a fix where I use a custom git-based plugin that I can use to get the resource file in there for phonegap build, since they don't support hooks there.

**Comment by gilvandev** [comment]
I have the same problem...but with res:// file too.

**Comment by mix3d** [comment]
@gilvandev, where is your file located, in the `/platforms/android/` project? Are you familiar with the structure of Android projects? When you specify with `res://`, that means the url needs to point to a file in the `project_root/res/` folder, but also split into the various DPI resolutions folders (drawable-xhdpi, drawable-hdpi, etc).

I have a script hook to copy my icon into the correct folders, that runs on the `after_platform_add` cordova hook, to copy them from my www folder into the correct res folder. 

TLDR: `res://` won't find files in the `www` folder

**Comment by ibnbd** [comment]
@mix3d my `platforms/android/res/` showing as bellow :

![screen2](https://cloud.githubusercontent.com/assets/2509519/14372665/85d27426-fd64-11e5-9bd9-2b3b6967231e.png)

can you please help me point the right file?

Thanks in advance

**Comment by iamola** [comment]
I have this same issue.
I have tried both res://icon.png and res://icon, along with file://img/icon and file://img/icon.png

I have icon.png images placed in the /www/ /www/img/ and /www/res/ folder. 
Please help.

**Comment by pgcd** [comment]
It took me a loooooong time but I think I have it completely under control! 
My relevant notification options are:
`{ 
icon: 'file://assets/icon/notification_icon.png', 
smallIcon: 'res://ic_notification_icon_large' 
}`
The first is a 128x128 icon in **/src/assets/icon/notification_icon.png** (it's the one you see when you pull down the notification area) - I have tried several other sizes and this seemed the correct one.
The second (the res:// one) is actually a set of icons that I prepared with http://romannurik.github.io/AndroidAssetStudio/ using asset size = 24dp. The result is a handy zip containing a single "res" folder - just copy this folder in your **/platforms/android** folder so that it merges with the existing "res" folder there, and voilà, all the properly resized icons are there.

**Comment by rwillett** [comment]
Is this still an issue?

**Comment by rwillett** [comment]
Closed as no update from OP.

**Comment by sekar-worx** [comment]
@mix3d, Hey, Can you please give me the sample cordova hook code for paste the images to drawable folder ?

**Comment by marcovtwout** [comment]
For pasting the images to the drawable folder, see: https://github.com/katzer/cordova-plugin-local-notifications/issues/1398
